android/src/main/java/com/github/yamill/orientation/OrientationModule.kt

```diff
@@ -0,0 +1,120 @@
+package com.github.yamill.orientation
+
+import android.annotation.SuppressLint
+import android.content.pm.ActivityInfo
+import android.content.res.Configuration
+import com.facebook.react.bridge.Callback
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.bridge.ReactContextBaseJavaModule
+import com.facebook.react.bridge.ReactMethod
+import com.github.yamill.orientation.listeners.OrientationAutoRotateListener
+import com.github.yamill.orientation.listeners.OrientationConfigListener
+
+
+class OrientationModule(reactContext: ReactApplicationContext) :
+    ReactContextBaseJavaModule(reactContext) {
+
+    private var autoRotateEnabled = false
+    private var autoRotateIgnored = false
+
+    override fun getName() = "Orientation"
+
+    override fun getConstants(): Map<String, Any?> {
+        val orientationInt = reactApplicationContext.resources.configuration.orientation
+        val orientation = getOrientationString(orientationInt)
+
+        return mapOf("initialOrientation" to orientation)
+    }
+
+    init {
+        reactContext.addLifecycleEventListener(
+            OrientationAutoRotateListener(reactContext) { autoRotateEnabled ->
+                this.autoRotateEnabled = autoRotateEnabled
+                this.maybeLockToPortrait()
+            }
+        )
+        reactContext.addLifecycleEventListener(
+            OrientationConfigListener(reactContext) {
+                currentActivity
+            }
+        )
+    }
+
+    @Suppress("unused")
+    @ReactMethod
+    fun getOrientation(callback: Callback) {
+        val orientationInt = reactApplicationContext.resources.configuration.orientation
+        val orientation = getOrientationString(orientationInt)
+        if (orientation === "null") {
+            callback.invoke(orientationInt, null)
+        } else {
+            callback.invoke(null, orientation)
+        }
+    }
+
+    @Suppress("unused")
+    @ReactMethod
+    fun ignoreAutoRotate(ignoreAutoRotate: Boolean) {
+        this.autoRotateIgnored = ignoreAutoRotate
+        this.maybeLockToPortrait()
+    }
```

Comment:
New API surface to disable the default behavior of respecting the auto-rotate setting.